ownCloud/Gallery/GalleryHostViewController.swift

```diff
@@ -0,0 +1,208 @@
+//
+//  AlternativePageViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 14/02/2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+
+class GalleryHostViewController: UIPageViewController {
+
+	// MARK: - Instance Variables
+	weak private var core: OCCore?
+	private var selectedItem: OCItem
+	private var items: [OCItem]?
+	private var query: OCQuery
+	private weak var viewControllerToTansition: DisplayViewController?
+
+	private var itemsFilter: OCQueryFilter = {
+		var itemFilterHandler: OCQueryFilterHandler = { (_, _, item) -> Bool in
+			if let item = item {
+				if item.type == .collection {return false}
+			}
+			return true
+		}
+
+		return OCQueryFilter(handler: itemFilterHandler)
+	}()
+
+	// MARK: - Init & deinit
+	init(core: OCCore, selectedItem: OCItem, query: OCQuery) {
+		self.core = core
+		self.selectedItem = selectedItem
+		self.query = query
+
+		super.init(transitionStyle: .scroll, navigationOrientation: .horizontal, options: nil)
+
+		query.addObserver(self, forKeyPath: "hasChangesAvailable", options: [.initial, .new], context: nil)
+		query.addFilter(itemsFilter, withIdentifier: "items-filter")
```

Comment:
As far as I can see looking through the changes on GitHub, you're reusing the `OCQuery` instance from the `ClientQueryViewController` instead of creating a new `OCQuery`, so that adding a filter to only show files would also alter what can be seen in the `ClientQueryViewController`.

An alternative would be to perform that filtering in `observeValue(forKeyPath:…)` in the completionHandler of the `query.requestChangeSet` call.

Also, regarding working KVO: you may need to first merge in the latest changes from `ios-app/master` to get the latest SDK version which should properly report changes via the `hasChangesAvailable` property (IIRC all but the latest had cases where it did not).